### ISSUE TITLE
bug: Class (str, enum.Enum)::__getitem__ expects 2 positional arguments.

### ISSUE THREAD

**Issue by Kamilcuk** [microsoft/pyright#6422]
**Describe the bug**
Enum.enum operator[] requires one argument, not two.

**Code or Screenshots**

```python
import enum
class A(str, enum.Enum):
    pass
A["a"]
```

Results in:

```
1.py:4:1 - error: Expected 1 more positional argument (reportGeneralTypeIssues)
```

**VS Code extension or command-line**
Command line, pyright, installed with npm on archlinux latest.

```
$ npm install -g pyright@1.1.335
changed 1 package in 2s
$ pyright ./1.py
.../1.py
  .../1.py:4:1 - error: Expected 1 more positional argument (reportGeneralTypeIssues)
1 error, 0 warnings, 0 informations 
$ npm install -g pyright@1.1.334
changed 1 package in 2s
$ pyright ./1.py
0 errors, 0 warnings, 0 informations 
```

Worked fine in 1.1.334, error in 1.1.335. Inheriting from `str` and `enum.Enum` is the way to do `StrEnum` in python 3.7.

**Comment by Kamilcuk** [comment]
Duplicate of https://github.com/microsoft/pyright/issues/6387
